src/hooks/useFirestoreConnect.ts

```diff
@@ -11,17 +12,36 @@ import { ValidFirestoreKeys } from "types/Firestore";
  */
 export type Defined<T> = T & Exclude<T, undefined>;
 
+/**
+ * This type allows us to avoid bugs when doc is not specified, and
+ * automagically constrain the storeAs parameter to keys that exist
+ * within ValidFirestoreKeys, ensuring that we can't forget to define
+ * it in our types when using functions that rely on this type
+ * (eg. useFirestoreConnect)
+ *
+ * @see ValidFirestoreKeys
+ * @see useFirestoreConnect
+ * @see ReduxFirestoreQuerySetting
+ */
+export interface SparkleRFQSetting extends ReduxFirestoreQuerySetting {
```

Comment:
Why the rename? Guessing because we already had a `Q` in `SparkleRFQ`?

We're extending `ReduxFirestoreQuerySetting`, which relates to a query. `SparkleRFQSetting` is less obvious in intent without additional mental process IMO. If we want both, something like `ReduxFirestoreQuerySetting` maybe?

Not sure it's name even really needs to mirror the type it's extending though, so maybe just `SparkleFirestoreQuery` or something?